src/client.rs

```diff
@@ -14,36 +16,35 @@ pub struct EigenClient {
 }
 
 impl EigenClient {
-    pub async fn new(config: EigenConfig, secrets: EigenSecrets) -> anyhow::Result<Self> {
-        let private_key = SecretKey::from_str(secrets.private_key.0.expose_secret().as_str())
-            .map_err(|e| anyhow::anyhow!("Failed to parse private key: {}", e))?;
+    pub async fn new(config: EigenConfig, secrets: EigenSecrets) -> Result<Self, EigenClientError> {
```

Comment:
Is using a single huge crate-level error enum like `EigenClientError` the typical way to go in rust? Reading https://www.reddit.com/r/rust/comments/1bb7dco/error_handling_goodbest_practices/ it seems like positions are mitigated on this issue.

It feels a bit weird to me. Wouldn't it be better to have a specific error enum per method? For eg, one guy says:
> As a rule of thumb, yes, errors should be specific to the operation. This ensures that client code doesn't have to be littered with unreachable!() statements for failure cases that are known to be impossible for that operation.

Reply:
Having a single error type that encapsulates the possible errors from a library makes it easier to use (specially if you use several modules from the same library together). 

We did a small refactor of the error types since your last review. So feel free to take a look again.